Ryecharm interfering with "go to declaration and usages"

### What happened?

Ryecharm interferes with "go to declaration and usages" as it only allows going to declaration and stops from opening the context window for usages or auto-moves in case only 1 usage exists:

In all cases, we only move to the definition, not to the usage. To open the list shown in the image, I have to click the mouse on usages as shown here:

<img width="552" height="171" alt="Image" src="https://github.com/user-attachments/assets/cbf5883a-406b-4c55-9512-c62ec724b532" />

This leads to me being able to navigate to the definition but never back to its usages. While this is mainly an annoyance as I use gd for this purpose with IdeaVim turned on, this also happens with vim entirely disabled, and having remapped "the go to definition or usages" in order to ensure this is not a weird keybinding issue. Additionally I have gone through all other "go to" functionalities in the dropdown menu, in case any of those now move to usages, however I never move to the usage or open the context window shown in the first image.

Disabling plugins leads me to Ryecharm as the main issue. See pycharm issue for initial thread: https://youtrack.jetbrains.com/issue/PY-89014/Go-to-declaration-or-usages-only-goes-to-declaration-but-not-to-usages

When I turn off ruff/ty by setting it to the unavailable option (not CLI or LSP) the problem remains. What data would you like me to provide to help out with this, I added in the pycharm log files, and have confirmed it is soley ryecharm and no other plugin. 




### Steps to reproduce

Steps to Reproduce 

-- Open a project in the IDE, install ryecharm, uv, ty, ruff and enable plugins.
--Define function, class, method
--Create a function call or object instantiation

--Hover over the definition and press the shortcut for "Go to declaration or usages" or gd using IdeaVim plugin (which uses the same functionality)

Expected Result 
-- In the case that there is no usage, stay at the definition.
-- In the case that there is a single use, move to it.
-- In the case that there are multiple uses, I expect the following menu to show up:

<img width="1235" height="189" alt="Image" src="https://github.com/user-attachments/assets/4f2d996a-2d3b-46eb-8d83-17e453ee0cb9" />

In the log file of ty, i removed some path information as i don't want to share my full paths, hope that is fine!

### Relevant log output or stack trace

```text
Ty output log:
2026-04-10 12:45:31.338136600 DEBUG main ty_server::server: Initialization options: InitializationOptions { log_level: Some(Trace), log_file: Some(), options: ClientOptions { global: GlobalOptions { diagnostic_mode: None, experimental: None }, workspace: WorkspaceOptions { disable_language_services: None, inlay_hints: None, python_extension: None }, unknown: {} } }
2026-04-10 12:45:31.341231900  INFO main ty_server::server: Version: 0.0.1-alpha.25 (3abd4c968 2025-10-29)
2026-04-10 12:45:31.342300200 DEBUG ty:main ty_server::server::main_loop: Requesting workspace configuration for workspaces
    2026-04-10 12:45:31.342915800 DEBUG ty:main client_response{id=0 method="workspace/configuration"}: ty_server::server::main_loop: Received workspace configurations, initializing workspaces
2026-04-10 12:45:31.343053700 DEBUG ty:main ty_server::session: Initializing workspace `file://EcoOptimizer`
2026-04-10 12:45:31.343185900  WARN ty:main ty_server::session: Received unknown options for workspace `file:/EcoOptimizer`: {
    "completions": {
        "autoImport": true
    },
    "showSyntaxErrors": true
}
2026-04-10 12:45:31.343313300 DEBUG ty:main ty_project::metadata: Searching for a project in ''
    2026-04-10 12:45:31.343708900 DEBUG ty:main ty_project::metadata::pyproject: Resolving requires-python constraint: `>=3.11`
2026-04-10 12:45:31.343816400 DEBUG ty:main ty_project::metadata::pyproject: Resolved requires-python constraint to: 3.11
2026-04-10 12:45:31.343909000 DEBUG ty:main ty_project::metadata: Found project at ''
2026-04-10 12:45:31.344020600 DEBUG ty:main ty_project::metadata::configuration_file: Searching for a user-level configuration at `\AppData\Roaming\ty\ty.toml`
2026-04-10 12:45:31.344746600  INFO ty:main ty_project::metadata::options: Defaulting to python-platform `win32`
2026-04-10 12:45:31.344999400 DEBUG ty:main ty_python_semantic::site_packages: Attempting to parse virtual environment metadata at '\.venv\pyvenv.cfg'
2026-04-10 12:45:31.345305300 DEBUG ty:main ty_python_semantic::site_packages: Searching for site-packages directory in sys.prefix \.venv
2026-04-10 12:45:31.345474100 DEBUG ty:main ty_python_semantic::site_packages: Resolved site-packages directories for this virtual environment are: SitePackagesPaths({"EcoOptimizer\\.venv\\Lib\\site-packages"})
2026-04-10 12:45:31.345832000 DEBUG ty:main ty_python_semantic::site_packages: Searching for real stdlib directory in sys.prefix E:\Python
2026-04-10 12:45:31.346012900 DEBUG ty:main ty_python_semantic::site_packages: Resolved real stdlib path for this virtual environment is: E:\Python\Lib
2026-04-10 12:45:31.346114800 DEBUG ty:main ty_python_semantic::module_resolver::resolver: Adding first-party search path `\SRC`
2026-04-10 12:45:31.346293600 DEBUG ty:main ty_python_semantic::module_resolver::resolver: Using vendored stdlib
2026-04-10 12:45:31.346606300 DEBUG ty:main ty_python_semantic::module_resolver::resolver: Adding site-packages search path `\.venv\Lib\site-packages`
2026-04-10 12:45:31.346777000  INFO ty:main ty_project::metadata::options: Python version: Python 3.11, platform: win32
2026-04-10 12:45:31.346894300 DEBUG ty:main ruff_db::files::file_root: Adding new file root '\.venv\Lib\site-packages' of kind LibrarySearchPath
2026-04-10 12:45:31.347317400 DEBUG ty:main ruff_db::files::file_root: Adding new file root '' of kind Project
2026-04-10 12:45:31.347474100 DEBUG ty:main ty_python_semantic::module_resolver::resolver: Resolving dynamic module resolution paths
2026-04-10 12:45:31.348498000 DEBUG ty:main client_response{id=1 method="client/registerCapability"}: ty_server::session: Registered dynamic capabilities
2026-04-10 12:45:31.376148100 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\__init__.py` to project `ecooptimizer`
2026-04-10 12:45:31.376326200 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\WKK_Generator.py` to project `ecooptimizer`
2026-04-10 12:45:31.376442900 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\Stationary_battery.py` to project `ecooptimizer`
2026-04-10 12:45:31.376542700 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\Smartbox.py` to project `ecooptimizer`
2026-04-10 12:45:31.376651100 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\RunSettings.py` to project `ecooptimizer`
2026-04-10 12:45:31.376768200 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\PVSystem.py` to project `ecooptimizer`
2026-04-10 12:45:31.376876600 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\Project.py` to project `ecooptimizer`
2026-04-10 12:45:31.376977600 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\Mobile_battery.py` to project `ecooptimizer`
2026-04-10 12:45:31.377092000 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\managers\backend_manager.py` to project `ecooptimizer`
2026-04-10 12:45:31.377193900 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\BuildingSituation.py` to project `ecooptimizer`
2026-04-10 12:45:31.377298300 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\managers\utils.py` to project `ecooptimizer`
2026-04-10 12:45:31.377404800 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\Battery_MobileBattery.py` to project `ecooptimizer`
2026-04-10 12:45:31.377500600 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\Consumption.py` to project `ecooptimizer`
2026-04-10 12:45:31.377594000 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\Finances.py` to project `ecooptimizer`
2026-04-10 12:45:31.377704100 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\Case_report.py` to project `ecooptimizer`
2026-04-10 12:45:31.377810100 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\Lighting.py` to project `ecooptimizer`
2026-04-10 12:45:31.377907600 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\managers\new_situation_wrapper.py` to project `ecooptimizer`
2026-04-10 12:45:31.378001000 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\managers\__init__.py` to project `ecooptimizer`
2026-04-10 12:45:31.378089900 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\Main_util.py` to project `ecooptimizer`
2026-04-10 12:45:31.378278100 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\Electric_heating.py` to project `ecooptimizer`
2026-04-10 12:45:31.378461800 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\input_class_wrapper.py` to project `ecooptimizer`
2026-04-10 12:45:31.378551600 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\Cooling.py` to project `ecooptimizer`
2026-04-10 12:45:31.378645400 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\EV_Charging.py` to project `ecooptimizer`
2026-04-10 12:45:31.378736300 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\components\base_adapter.py` to project `ecooptimizer`
2026-04-10 12:45:31.378869400 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\components\base_component.py` to project `ecooptimizer`
2026-04-10 12:45:31.378963500 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ty_project: Adding file `\SRC\EcoOptimizer\backend\components\__init__.py` to project `ecooptimizer`
2026-04-10 12:45:31.379072600 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Opening file `\SRC\EcoOptimizer\backend\Mobile_battery.py`
2026-04-10 12:45:31.379165800 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Take open project files
2026-04-10 12:45:31.379277700 DEBUG ty:main notification{method="textDocument/didOpen"}:set_open_files{open_files={file(Id(1007))}}: ty_project: Set open project files (count: 1)
2026-04-10 12:45:31.382519500 DEBUG ThreadId(50) ty_project::walk: Skipping directory '\.venv' because it is excluded by a default or `src.exclude` pattern
2026-04-10 12:45:31.382984900 DEBUG ThreadId(50) ty_project::walk: Skipping directory '\.ruff_cache' because it is excluded by a default or `src.exclude` pattern
2026-04-10 12:45:31.383299200 DEBUG ThreadId(41) ty_project::walk: Skipping directory '\.git' because it is excluded by a default or `src.exclude` pattern
2026-04-10 12:45:31.388132500  INFO ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(1007))}:Project::index_files{project=ecooptimizer}: ty_project: Indexed 81 file(s) in 0.009s
2026-04-10 12:45:31.388492400 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(1007))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\Mobile_battery.py'
2026-04-10 12:45:31.437766500 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ruff_db::files: Updating the revision of `\SRC\EcoOptimizer\frontend\master_page_layout.py`
2026-04-10 12:45:31.438145500 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Opening file `\SRC\EcoOptimizer\frontend\master_page_layout.py`
2026-04-10 12:45:31.438267300 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Take open project files
2026-04-10 12:45:31.438387000 DEBUG ty:main notification{method="textDocument/didOpen"}:set_open_files{open_files={file(Id(1007)), file(Id(2402))}}: ty_project: Set open project files (count: 2)
2026-04-10 12:45:31.438563000 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(2402))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\frontend\master_page_layout.py'
2026-04-10 12:45:31.441472200 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ruff_db::files: Updating the revision of `\SRC\EcoOptimizer\backend\Consumption.py`
2026-04-10 12:45:31.441604600 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Opening file `\SRC\EcoOptimizer\backend\Consumption.py`
2026-04-10 12:45:31.441717900 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Take open project files
2026-04-10 12:45:31.441820400 DEBUG ty:main notification{method="textDocument/didOpen"}:set_open_files{open_files={file(Id(1007)), file(Id(2402)), file(Id(2000))}}: ty_project: Set open project files (count: 3)
2026-04-10 12:45:31.442579900 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(2000))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\Consumption.py'
2026-04-10 12:45:31.505348300 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(2000))}: ty_python_semantic::module_resolver::resolver: Module `odf.opendocument` not found in search paths
2026-04-10 12:45:31.506138300 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(2000))}: ty_python_semantic::module_resolver::resolver: Module `pyxlsb.workbook` not found in search paths
2026-04-10 12:45:31.576470700  INFO ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(2000))}: ty_python_semantic::types: Checking file `\SRC\EcoOptimizer\backend\Consumption.py` took more than 100ms (133.6326ms)
2026-04-10 12:45:31.576791400 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ruff_db::files: Updating the revision of `\SRC\EcoOptimizer\backend\components\base_component.py`
2026-04-10 12:45:31.577402700 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Opening file `\SRC\EcoOptimizer\backend\components\base_component.py`
2026-04-10 12:45:31.577517900 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Take open project files
2026-04-10 12:45:31.577624300 DEBUG ty:main notification{method="textDocument/didOpen"}:set_open_files{open_files={file(Id(1007)), file(Id(1009)), file(Id(2000)), file(Id(2402))}}: ty_project: Set open project files (count: 4)
2026-04-10 12:45:31.577853100 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(1009))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\components\base_component.py'
2026-04-10 12:45:31.581082200 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ruff_db::files: Updating the revision of `\SRC\EcoOptimizer\backend\Stationary_battery.py`
2026-04-10 12:45:31.581287600 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Opening file `\SRC\EcoOptimizer\backend\Stationary_battery.py`
2026-04-10 12:45:31.581379900 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Take open project files
2026-04-10 12:45:31.581484600 DEBUG ty:main notification{method="textDocument/didOpen"}:set_open_files{open_files={file(Id(1007)), file(Id(1009)), file(Id(2000)), file(Id(2402)), file(Id(1002))}}: ty_project: Set open project files (count: 5)
2026-04-10 12:45:31.581738100 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(1002))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\Stationary_battery.py'
2026-04-10 12:45:31.584188800 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ruff_db::files: Updating the revision of `\SRC\EcoOptimizer\backend\components\base_adapter.py`
2026-04-10 12:45:31.584313700 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Opening file `\SRC\EcoOptimizer\backend\components\base_adapter.py`
2026-04-10 12:45:31.584423800 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Take open project files
2026-04-10 12:45:31.584523400 DEBUG ty:main notification{method="textDocument/didOpen"}:set_open_files{open_files={file(Id(1007)), file(Id(1009)), file(Id(2000)), file(Id(1008)), file(Id(2402)), file(Id(1002))}}: ty_project: Set open project files (count: 6)
2026-04-10 12:45:31.584777100 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(1008))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\components\base_adapter.py'
2026-04-10 12:45:31.590135900 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ruff_db::files: Updating the revision of `\SRC\EcoOptimizer\frontend\components_callbacks\input_components.py`
2026-04-10 12:45:31.590389200 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Opening file `\SRC\EcoOptimizer\frontend\components_callbacks\input_components.py`
2026-04-10 12:45:31.590497900 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Take open project files
2026-04-10 12:45:31.590622800 DEBUG ty:main notification{method="textDocument/didOpen"}:set_open_files{open_files={file(Id(1007)), file(Id(1009)), file(Id(2000)), file(Id(1008)), file(Id(2402)), file(Id(1002)), file(Id(2005))}}: ty_project: Set open project files (count: 7)
2026-04-10 12:45:31.591041200 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(2005))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\frontend\components_callbacks\input_components.py'
2026-04-10 12:45:31.635265700 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ruff_db::files: Updating the revision of `\SRC\EcoOptimizer\backend\Battery_MobileBattery.py`
2026-04-10 12:45:31.635815000 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Opening file `\SRC\EcoOptimizer\backend\Battery_MobileBattery.py`
2026-04-10 12:45:31.635941500 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Take open project files
2026-04-10 12:45:31.636061800 DEBUG ty:main notification{method="textDocument/didOpen"}:set_open_files{open_files={file(Id(1009)), file(Id(1c00)), file(Id(2000)), file(Id(1008)), file(Id(1007)), file(Id(2402)), file(Id(1002)), file(Id(2005))}}: ty_project: Set open project files (count: 8)
2026-04-10 12:45:31.636771500 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(1c00))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\Battery_MobileBattery.py'
2026-04-10 12:45:31.649684300 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ruff_db::files: Updating the revision of `\SRC\EcoOptimizer\backend\managers\backend_manager.py`
2026-04-10 12:45:31.650154400 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Opening file `\SRC\EcoOptimizer\backend\managers\backend_manager.py`
2026-04-10 12:45:31.650273000 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Take open project files
2026-04-10 12:45:31.650392400 DEBUG ty:main notification{method="textDocument/didOpen"}:set_open_files{open_files={file(Id(1009)), file(Id(1c00)), file(Id(2000)), file(Id(1008)), file(Id(1007)), file(Id(1400)), file(Id(2402)), file(Id(1002)), file(Id(2005))}}: ty_project: Set open project files (count: 9)
2026-04-10 12:45:31.650705800 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(1400))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\managers\backend_manager.py'
2026-04-10 12:45:31.659617800 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ruff_db::files: Updating the revision of `\SRC\EcoOptimizer\backend\Lighting.py`
2026-04-10 12:45:31.659795400 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Opening file `\SRC\EcoOptimizer\backend\Lighting.py`
2026-04-10 12:45:31.659906100 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Take open project files
2026-04-10 12:45:31.660026400 DEBUG ty:main notification{method="textDocument/didOpen"}:set_open_files{open_files={file(Id(1009)), file(Id(1c00)), file(Id(2000)), file(Id(1008)), file(Id(1007)), file(Id(2800)), file(Id(1400)), file(Id(2402)), file(Id(1002)), file(Id(2005))}}: ty_project: Set open project files (count: 10)
2026-04-10 12:45:31.660356000 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(2800))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\Lighting.py'
2026-04-10 12:45:31.667159100 DEBUG ty:main notification{method="textDocument/didOpen"}:apply_changes: ruff_db::files: Updating the revision of `\SRC\EcoOptimizer\oldConsumption.py`
2026-04-10 12:45:31.667423100 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Opening file `\SRC\EcoOptimizer\oldConsumption.py`
2026-04-10 12:45:31.667525200 DEBUG ty:main notification{method="textDocument/didOpen"}: ty_project: Take open project files
2026-04-10 12:45:31.667634700 DEBUG ty:main notification{method="textDocument/didOpen"}:set_open_files{open_files={file(Id(1009)), file(Id(1c00)), file(Id(2000)), file(Id(1008)), file(Id(2c01)), file(Id(1007)), file(Id(2800)), file(Id(1400)), file(Id(2402)), file(Id(1002)), file(Id(2005))}}: ty_project: Set open project files (count: 11)
2026-04-10 12:45:31.668887700 DEBUG ty:main notification{method="textDocument/didOpen"}:check_file{file=file(Id(2c01))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\oldConsumption.py'
2026-04-10 12:45:31.697703800 DEBUG ty:worker:3 request{id="5" method="textDocument/diagnostic"}:check_file{file=file(Id(2800))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\Lighting.py'
2026-04-10 12:45:31.697738000 DEBUG ty:worker:2 request{id="4" method="textDocument/diagnostic"}:check_file{file=file(Id(1002))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\Stationary_battery.py'
2026-04-10 12:45:31.697893200 DEBUG ty:worker:0 request{id="3" method="textDocument/diagnostic"}:check_file{file=file(Id(2005))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\frontend\components_callbacks\input_components.py'
2026-04-10 12:45:31.698050800 DEBUG ty:worker:1 request{id="2" method="textDocument/diagnostic"}:check_file{file=file(Id(1400))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\managers\backend_manager.py'
2026-04-10 12:45:31.701896300 DEBUG ty:worker:2 request{id="6" method="textDocument/diagnostic"}:check_file{file=file(Id(2000))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\Consumption.py'
2026-04-10 12:45:31.703913300 DEBUG ty:worker:1 request{id="7" method="textDocument/diagnostic"}:check_file{file=file(Id(1c00))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\Battery_MobileBattery.py'
2026-04-10 12:45:31.705344100 DEBUG     ty:main notification{method="workspace/didChangeWatchedFiles"}: ty_server::server::api::notifications::did_change_watched_files: Applying changes to ``
2026-04-10 12:45:31.707494900 DEBUG     ty:main notification{method="workspace/didChangeWatchedFiles"}:check_file{file=file(Id(2402))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\frontend\master_page_layout.py'
2026-04-10 12:45:31.707815400 DEBUG     ty:main notification{method="workspace/didChangeWatchedFiles"}:check_file{file=file(Id(1008))}: ty_python_semantic::types: Checking file '\SRC\EcoOptimizer\backend\components\base_adapter.py'
2026-04-10 12:45:31.711985900  WARN     ty:main notification{method="workspace/didChangeWatchedFiles"}:check_file{file=file(Id(1007))}: salsa::function::execute: Propagating panic for cycle head that panicked in an earlier execution in that revision
2026-04-10 12:45:31.745735800 ERROR        main ty_server::server: panicked at C:\Users\runneradmin\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\jod-thread-1.0.0\src\lib.rs:33:22:
called `Result::unwrap()` on an `Err` value: Any { .. }
0: <unknown>
1: <unknown>
2: <unknown>
3: <unknown>
4: <unknown>
5: <unknown>
6: <unknown>
7: <unknown>
8: <unknown>
9: <unknown>
10: <unknown>
11: <unknown>
12: <unknown>
13: <unknown>
14: <unknown>
15: BaseThreadInitThunk
16: RtlUserThreadStart
```

### Operating system

Windows

Thanks. The gist of these errors is:

> `Caused by: java.lang.ClassCastException: class insyncwithfoo.ryecharm.configurations.ty.TYGlobalService cannot be cast to class insyncwithfoo.ryecharm.configurations.ty.TYGlobalService (insyncwithfoo.ryecharm.configurations.ty.TYGlobalService is in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @3da42dc3; insyncwithfoo.ryecharm.configurations.ty.TYGlobalService is in unnamed module of loader com.intellij.ide.plugins.cl.PluginClassLoader @26a0ef1d)`

```text
	at insyncwithfoo.ryecharm.configurations.ty.TYGlobalService$Companion.getInstance(Services.kt:57)
	at insyncwithfoo.ryecharm.configurations.ty.TYConfigurable.<init>(Configurables.kt:23)
	at com.intellij.serviceContainer.ComponentManagerImpl.findConstructorAndInstantiateClass(ComponentManagerImpl.kt:970)
	at com.intellij.serviceContainer.ComponentManagerImpl.doInstantiateClass(ComponentManagerImpl.kt:979)
```

For some reason, old service instances weren't disposed of properly, so when the plugin is loaded a second time by a different loader, `TYGlobalService.getInstance()` returns the lingering instance, which of course is not an instance of the new class. I'm not sure how or why this happened. I filed #218. It would be great if you could post a comment there with a minimal reproducible example (no pressure though, these errors are often very hard to reproduce, and you are busy too, I imagine).

----

Believe it or not, adding a new setting is trivial enough. You can download the build from [here](https://github.com/InSyncWithFoo/ryecharm/actions/runs/24252293561) (see also [manual install guide](https://www.jetbrains.com/help/pycharm/managing-plugins.html#install_plugin_from_disk)). Let me know if that works.

Thanks again for being an awesome user.

--- a/docs/configurations/ty.md
+++ b/docs/configurations/ty.md
@@ -101,6 +101,14 @@ language server setting.
 Default: `true`
 
 
+## Go to definition
+
+Whether PyCharm's <i>Go to Declaration</i>
+and related features should defer to ty.
+
+Default: `true`
+
+
 ## Log level
 
 The amount of logs the language server will emit.

--- a/src/main/kotlin/insyncwithfoo/ryecharm/configurations/ty/Configurations.kt
+++ b/src/main/kotlin/insyncwithfoo/ryecharm/configurations/ty/Configurations.kt
@@ -60,6 +60,8 @@ internal class TYConfigurations : DisplayableState(), Copyable {
     var completions by property(true)
     var completionsAutoImport by property(true)
     
+    var goToDefinition by property(true)
+    
     var logLevel by enum(LogLevel.INFO)
     var logFile by string(null)
 }

--- a/src/main/kotlin/insyncwithfoo/ryecharm/configurations/ty/Panels.kt
+++ b/src/main/kotlin/insyncwithfoo/ryecharm/configurations/ty/Panels.kt
@@ -81,6 +81,10 @@ private fun Row.completionsAutoImportInput(block: Cell<JBCheckBox>.() -> Unit) =
     checkBox(message("configurations.ty.completionsAutoImport.label")).apply(block)
 
 
+private fun Row.goToDefinitionInput(block: Cell<JBCheckBox>.() -> Unit) =
+    checkBox(message("configurations.ty.goToDefinition.label")).apply(block)
+
+
 private fun Row.logLevelInput(block: Cell<ComboBox<LogLevel>>.() -> Unit) =
     comboBox<LogLevel>().apply(block)
 
@@ -163,6 +167,11 @@ private fun TYPanel.makeComponent() = panel {
                 inlayHintsCallArgumentNamesInput { bindSelected(state::inlayHintsCallArgumentNames) }
                 overrideCheckbox(state::inlayHintsCallArgumentNames)
             }
+        }
+        
+        row {
+            goToDefinitionInput { bindSelected(state::goToDefinition) }
+            overrideCheckbox(state::goToDefinition)
         }
         
         separator()

--- a/src/main/kotlin/insyncwithfoo/ryecharm/ty/lsp/GoToDefinitionSupport.kt
+++ b/src/main/kotlin/insyncwithfoo/ryecharm/ty/lsp/GoToDefinitionSupport.kt
@@ -1,0 +1,6 @@
+package insyncwithfoo.ryecharm.ty.lsp
+
+import com.intellij.platform.lsp.api.customization.LspGoToDefinitionSupport
+
+
+internal class GoToDefinitionSupport : LspGoToDefinitionSupport()

--- a/src/main/kotlin/insyncwithfoo/ryecharm/ty/lsp/TYServerDescriptor.kt
+++ b/src/main/kotlin/insyncwithfoo/ryecharm/ty/lsp/TYServerDescriptor.kt
@@ -9,6 +9,8 @@ import com.intellij.platform.lsp.api.customization.LspCompletionDisabled
 import com.intellij.platform.lsp.api.customization.LspCustomization
 import com.intellij.platform.lsp.api.customization.LspDiagnosticsCustomizer
 import com.intellij.platform.lsp.api.customization.LspDiagnosticsDisabled
+import com.intellij.platform.lsp.api.customization.LspGoToDefinitionCustomizer
+import com.intellij.platform.lsp.api.customization.LspGoToDefinitionDisabled
 import com.intellij.platform.lsp.api.customization.LspInlayHintCustomizer
 import com.intellij.platform.lsp.api.customization.LspInlayHintDisabled
 import insyncwithfoo.ryecharm.common.logging.tyLogger
@@ -48,6 +50,12 @@ internal class TYServerDescriptor(project: Project, private val executable: Path
             get() = when (configurations.completions) {
                 true -> CompletionSupport()
                 else -> LspCompletionDisabled
+            }
+        
+        override val goToDefinitionCustomizer: LspGoToDefinitionCustomizer
+            get() = when (configurations.goToDefinition) {
+                true -> GoToDefinitionSupport()
+                else -> LspGoToDefinitionDisabled
             }
         
     }

--- a/src/main/resources/messages/ryecharm.properties
+++ b/src/main/resources/messages/ryecharm.properties
@@ -216,6 +216,8 @@ configurations.ty.inlayHintsCallArgumentNames.label = Call argument names
 configurations.ty.completions.label = Completions
 configurations.ty.completionsAutoImport.label = Auto-import
 
+configurations.ty.goToDefinition.label = Go to definition
+
 configurations.ty.logLevel.label = Log level:
 configurations.ty.logLevel.trace = Trace
 configurations.ty.logLevel.debug = Debug

--- a/src/test/kotlin/insyncwithfoo/ryecharm/configurations/ty/TYConfigurationsTest.kt
+++ b/src/test/kotlin/insyncwithfoo/ryecharm/configurations/ty/TYConfigurationsTest.kt
@@ -12,7 +12,7 @@ internal class TYConfigurationsTest : ConfigurationsTest<TYConfigurations>() {
     
     @Test
     fun `test shape`() {
-        doShapeTest(expectedSize = 14) {
+        doShapeTest(expectedSize = 15) {
             assertEquals(null, executable)
             assertEquals(null, configurationFile)
             
@@ -30,6 +30,8 @@ internal class TYConfigurationsTest : ConfigurationsTest<TYConfigurations>() {
             
             assertEquals(true, completions)
             assertEquals(true, completionsAutoImport)
+            
+            assertEquals(true, goToDefinition)
             
             assertEquals(LogLevel.INFO, logLevel)
             assertEquals(null, logFile)

